File: app/controllers/ConfigController.scala
package controllers

import javax.inject.Inject

import play.api.libs.json.Json
import play.api.mvc.{Action, Controller}
import services.ConfigService

class ConfigController @Inject()() extends Controller{

  def getInitialConfig = Action{
    Ok(Json.obj(
      "mutationPercentage"  -> ConfigService.getMutationPercentage,
      "generationSize"      -> ConfigService.getGenerationSize,
      "numberOfGenerations" -> ConfigService.getNumberOfGenerations
    ))
  }

  def setMutationPercentage(mutationPercentage: Double)  = Action{
    ConfigService.setMutationPercentage(mutationPercentage)
    Ok
  }

  def setMaxWeight(maxWeight: Int) = Action{
    ConfigService.setMaxWeight(maxWeight)
    Ok
  }

  def setGenerationSize(generationSize: Int) = Action{
    ConfigService.setGenerationSize(generationSize)
    Ok
  }

  def setNumberOfGenerations(numberOfGenerations: Int) = Action{
    ConfigService.setNumberOfGenerations(numberOfGenerations)
    Ok
  }

  def setUpperBound(upperBound: Int) = Action{
    ConfigService.setUpperBound(upperBound)
    Ok
  }

  def getUpperBound() = Action{ Ok(Json.toJson(ConfigService.getUpperBound))}

  def getMaxWeight() = Action{ Ok(Json.toJson(ConfigService.getMaxWeight))}
}

File: public/main/directives/controllers/configController.js
angular.module("SurvivalOfTheFittestApp").controller("configController", ['$scope', 'algorithmService', function ($scope, algorithmService) {

    algorithmService.getInitialConfig().then(function(response){
        $scope.generationSize = response.data.generationSize;
        $scope.mutationPercentage = response.data.mutationPercentage;
        $scope.numberOfGenerations = response.data.numberOfGenerations;
    });

    $scope.updateMutationPercentage = function(){
        algorithmService.setMutationPercentage($scope.mutationPercentage);
    };

    $scope.updateGenerationSize = function(){
        algorithmService.setGenerationSize($scope.generationSize);
    };

    $scope.updateNumberOfGenerations = function(){
        algorithmService.setNumberOfGenerations($scope.numberOfGenerations);
    };
}]);